src/Schema/Keywords/Required.php

```diff
@@ -52,23 +55,27 @@ public function validate($data, array $required) : void
                 if (! $propertyFound) {
                     // respect writeOnly/readOnly keywords
                     if ((
-                            $this->parentSchema->properties[$reqProperty]->writeOnly &&
-                            $this->validationDataType === \OpenAPIValidation\Schema\Validator::VALIDATE_AS_RESPONSE
+                            ($this->parentSchema->properties[$reqProperty]->writeOnly ?? false) &&
```

Comment:
small inline bugfix for unset fields or parent properties

Reply:
Good catch. That would have troubled us in the future.